src/panel/direction/DirectionPanel.jsx

```diff
@@ -213,25 +213,29 @@ export default class DirectionPanel extends React.Component {
       previewRoute={activePreviewRoute}
     />;
 
-    if (Device.isMobile()) {
-      return <div className="direction_panel_mobile">
-        <div className="itinerary_close_mobile" onClick={this.onClose}>
-          <span className="icon-chevron-left" />
-          {_('return', 'direction')}
-        </div>
-        {title}
-        {!activePreviewRoute && form}
-        {result}
-      </div>;
-    }
-
-    return <Panel
-      title={title}
-      close={this.onClose}
-      className="direction_panel"
-    >
-      {form}
-      {result}
-    </Panel>;
+    return <DeviceContext.Consumer>
+      {isMobile => {
+        if (isMobile) {
```

Comment:
I know you did not edited these lines in this PR, but in order to be more functional we could write :

```js
{isMobile => (isMobile
  ? <FirstComp />
  : <SecondComp />
)}
```